src/code/sys_rumble.c

```diff
@@ -0,0 +1,156 @@
+#include "global.h"
```

Comment:
Can make some file comments too: this is intermediate between padmgr and the external file, whereas the external file is how you're actually supposed to interact with this? I suppose this is sort of the scheduling/processing file, the other is the requesting file. There's also the question of where to put an explanatory block about how the system works, since Update is really doing all the work here.